CheckoutTests/Tokenisation/TokenDetailsFactoryTests.swift

```diff
@@ -8,20 +8,81 @@
 import XCTest
 @testable import Checkout
 
-// swiftlint:disable implicitly_unwrapped_optional
 final class TokenDetailsFactoryTests: XCTestCase {
-  private var subject: TokenDetailsFactory! = TokenDetailsFactory()
-
-  override func tearDown() {
-    subject = nil
-
-    super.tearDown()
-  }
-
-  func test_create() {
-    let tokenResponse = StubProvider.createTokenResponse()
-    let expectedTokenDetails = StubProvider.createTokenDetails(expiryDate: ExpiryDate(month: 6, year: 2055))
-
-    XCTAssertEqual(subject.create(tokenResponse: tokenResponse), expectedTokenDetails)
-  }
+    
+    func testPropertiesMapped() {
+        let testType = TokenRequest.TokenType.card
+        let testToken = "12345676sdgdh"
+        let testExpiresOn = "tomorrow"
+        let testExpiryMonth = 6
+        let testExpiryYear = 3011
+        let testScheme = "visa"
+        let testSchemeLocal = "mada"
```

Comment:
hardcoded string is fine, but if you want to actually write the proper value this should be visa or mastercard, see [mada test cards](https://www.checkout.com/docs/payments/payment-methods/mada#Mada_test_cards).

Reply:
The test is purely validating that TokenDetailsFactory is able to map correctly data from network object to SDK model. As network object actually handles String for these properties, there is an infinite number of permutations we could test for. That would defeat the purpose of testing that a given valid network object is parsed successfully to an SDK model.